src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -334,30 +340,74 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
     }
 
     @Override
-    public Map<String, Long> materializePositions(final String topicId, final SubscriptionBase.InitialPosition position)
+    public Map<String, Long> materializePositions(final EventType eventType, final Subscription subscription)
```

Comment:
I agree that it belongs to the domain layer. In my opinion it has to be used in database repository layer and service layer which works with them, but topic repository slightly different thing which abstracts messaging backend system and probably, it should know nothing about Subscriptions.